### PR TITLE
feat(outbox): use content-type in the metadata for projections item

### PR DESCRIPTION
**Pull request by luigirende**
# Description

This request it is linked to PR https://github.com/dapr/dapr/pull/8062. In fact that PR didn't cover if we are using the projection item to publish the event. In this request if the item has `outbox.projection = true` the contentType in the metadata will be take if it's not set in the request

PR cloned from https://github.com/dapr/dapr/pull/8132 for branch `release-1.14`

## Issue reference

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [ ] Code compiles correctly
- [ ] Created/updated tests
- [ ] Unit tests passing
- [ ] End-to-end tests passing
- [ ] Extended the documentation / Created issue in the <https://github.com/dapr/docs/> repo: dapr/docs#_[issue number]_
- [ ] Specification has been updated / Created issue in the <https://github.com/dapr/docs/> repo: dapr/docs#_[issue number]_
- [ ] Provided sample for the feature / Created issue in the <https://github.com/dapr/docs/> repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by luigirende** [comment]
[cicoyle](https://github.com/cicoyle) for this change created a new PR

File: pkg/runtime/pubsub/outbox.go [hunk 1]
```diff
@@ -178,6 +178,8 @@ func (o *outboxImpl) PublishInternal(ctx context.Context, stateStore string, ope
 
 				if proj.ContentType != nil {
 					contentType = *proj.ContentType
+				} else if ct, ok := sr.Metadata[metadata.ContentType]; ok {
```
Comment (antontroshin) [inline pkg/runtime/pubsub/outbox.go:181]: I think this was meant to be `proj.Metadata[metadata.ContentType]`

Comment (luigirende) [reply pkg/runtime/pubsub/outbox.go]: Yeah. sorry, to switch the PR to `release-1.14` I've mistaken to type. Fix now thanks

**Review by antontroshin — DISMISSED** [review]

**Comment by JoshVanL** [comment]
Thanks @luigirende, please include an [integration test](https://github.com/dapr/dapr/pull/8062/files#diff-22af11fc45cebb40e50126237806b7158e62220a9cf6aa0ab3cc607170306a43R1) for this change.

**Comment by luigirende** [comment]
> Thanks @luigirende, please include an [integration test](https://github.com/dapr/dapr/pull/8062/files#diff-22af11fc45cebb40e50126237806b7158e62220a9cf6aa0ab3cc607170306a43R1) for this change.

@JoshVanL I've just added the integration tests.

**Review by cicoyle — APPROVED** [review]
Thanks for adding the test and targeting the `release-1.14` branch @luigirende 🎉 lgtm, assuming tests pass

**Review by antontroshin — APPROVED** [review]

**Pull request merged**